crates/rpc/api/src/lib.rs

```diff
@@ -1,18 +1,47 @@
 //! Macro trait def for the `strata_` RPC namespace using jsonrpsee.
 use bitcoin::Txid;
 use jsonrpsee::{core::RpcResult, proc_macros::rpc};
-use strata_db::types::L1TxStatus;
+use strata_db::types::{L1TxEntry, L1TxStatus};
 use strata_primitives::bridge::{OperatorIdx, PublickeyTable};
 use strata_rpc_types::{
     types::{RpcBlockHeader, RpcClientStatus, RpcL1Status},
-    HexBytes, HexBytes32, L2BlockStatus, RpcBridgeDuties, RpcCheckpointInfo, RpcDepositEntry,
-    RpcExecUpdate, RpcSyncStatus,
+    HexBytes, HexBytes32, L2BlockStatus, RpcBridgeDuties, RpcChainState, RpcCheckpointInfo,
+    RpcDepositEntry, RpcExecUpdate, RpcL2Block, RpcSyncStatus,
+};
+use strata_state::{
+    client_state::ClientState, id::L2BlockId, operation::ClientUpdateOutput, sync_event::SyncEvent,
 };
-use strata_state::{id::L2BlockId, operation::ClientUpdateOutput, sync_event::SyncEvent};
-
 #[cfg_attr(not(feature = "client"), rpc(server, namespace = "strata"))]
 #[cfg_attr(feature = "client", rpc(server, client, namespace = "strata"))]
 pub trait StrataApi {
+    /// Get the block by its id
+    #[method(name = "getBlockById")]
+    async fn get_block_by_id(&self, block_id: L2BlockId) -> RpcResult<Option<RpcL2Block>>;
+
+    /// Get blocks at a certain height
+    #[method(name = "getBlocksAtIdx")]
+    async fn get_blocks_at_idx(&self, idx: u64) -> RpcResult<Vec<HexBytes32>>;
+
+    /// Get chain tip block id
+    #[method(name = "blockNumber")]
+    async fn block_number(&self) -> RpcResult<u64>;
```

Comment:
Duplicates `tip_slot` from the current chainstate, don't need a RPC just for this.